src/rules/expect.ts

```diff
@@ -480,7 +583,56 @@ function getExpectTypeFailures(
 
     ts.forEachChild(node, iterate);
   });
-  return { unmetExpectations, unusedAssertions: typeAssertions.keys() };
+
+  const twoSlashFailureLines: number[] = [];
+  if (twoSlashAssertions.length) {
+    for (const assertion of twoSlashAssertions) {
+      const { position, expected } = assertion;
+      if (position === -1) {
+        // special case for a twoslash assertion on line 1.
+        twoSlashFailureLines.push(0);
+        continue;
+      }
+      const node = getNodeAtPosition(sourceFile, position);
+      if (!node) {
+        twoSlashFailureLines.push(sourceFile.getLineAndCharacterOfPosition(position).line);
+        continue;
+      }
+
+      const qi = languageService.getQuickInfoAtPosition(sourceFile.fileName, node.getStart());
+      if (!qi || !qi.displayParts) {
+        twoSlashFailureLines.push(sourceFile.getLineAndCharacterOfPosition(position).line);
+        continue;
+      }
+      const actual = qi.displayParts.map((dp) => dp.text).join('');
+      if (!matchModuloWhitespace(actual, expected)) {
+        unmetExpectations.push({ assertion: { assertionType: 'twoslash', ...assertion }, node, actual });
+      }
+    }
+  }
+
+  return { unmetExpectations, unusedAssertions: [...twoSlashFailureLines, ...typeAssertions.keys()] };
+}
+
+function getNodeAtPosition(sourceFile: ts.SourceFile, position: number): ts.Node | null {
+  let candidate: ts.Node | null = null;
```

Comment:
[Nitpick] Any reason to prefer `null` over `undefined`? 

```suggestion
function getNodeAtPosition(sourceFile: ts.SourceFile, position: number): ts.Node | undefined {
  let candidate: ts.Node | undefined;
```